sqlglot/generator.py

```diff
@@ -2379,7 +2379,11 @@ def function_fallback_sql(self, expression: exp.Func) -> str:
             elif arg_value is not None:
                 args.append(arg_value)
 
-        return self.func(expression.sql_name(), *args)
+        name = expression.sql_name()
```

Comment:
this should be called outside the block

so if self.normalize_functions... else

Reply:
added an `else` block to set it there